### ISSUE TITLE
Add constructor with accepting a single parameter to initalize all components to vectors

### ISSUE THREAD

**Issue by Chaosus** [godotengine/godot-proposals#3884]
### Describe the project you are working on

Any project

### Describe the problem or limitation you are having in your project

It's a regular nuisance with almost any project for me.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I want to propose a way to easily initialize all members in the vector in the constructor.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Currently, if you need to initialize a vector with all components equal to 0.5 you need to write:

```
var v = Vector3(0.5, 0.5, 0.5)
```
I want to write it like:
```
var v = Vector3(0.5)
``` 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Well, a few lines of code are not applicable for this case - it's a single line in both cases.

### Is there a reason why this should be core and not an add-on in the asset library?

Core change

**Comment by Calinou** [comment]
You can already use `var v = Vector3.ONE * 0.5`.

**Comment by Chaosus** [comment]
Yeah, I just leave it as a proposal - if users like it - it may be considered for implementation.

**Comment by Mickeon** [comment]
> You can already use var v = Vector3.ONE * 0.5 to achieve this
> To be honest var v = Vector3.ONE * 0.5 looks a bit unintuitive for me.

I always kept thinking about a proposal like this on the back of my head for the many times I needed this. However, I personally find `Vector3.ONE * 0.5` to be more readable. There's less numbers clumped together, and it definitely helps somebody reading back on the code in the long run.

That's not to say I'm against adding it.

**Comment by Mickeon** [comment]
To add to it, I find there could be some embarrassment of choice due of preference between `Vector2(number)` and `Vector3.ONE * number`, but of the former was optimised by the compiler to equal to `Vector2(number)` and ditch the multiplication, it could definitely close the gap.

**Comment by tokengamedev** [comment]
IMO 
```gdscript
var v := Vector3(0.5)
```
is not equivalent to
```gdscript
var v := 0.5 * Vector3.ONE 
```
in the sense that in the first you just create a vector and in the second you are creating and multiplying. 

In the second, It always creates a doubt in my mind why I am multiplying.

**Comment by tokengamedev** [comment]
I want to add one more scenario, which in my case the most common one for my projects at least.
```gdscript
var image_scale = window_size.y / image.texture.size.y
image.scale = Vector2(image_scale, image_scale)
```
I hate writing this code, as it always comes up why I can't write like this:

```gdscript
image.scale = Vector2(window_size.y / image.texture.size.y)
```
